compiler/rustc_middle/src/ty/print/pretty.rs

```diff
@@ -1830,96 +1834,136 @@ impl<F: fmt::Write> FmtPrinter<'_, 'tcx, F> {
+            let mut region_map: BTreeMap<ty::BoundRegionKind, ty::Region<'tcx>> = BTreeMap::new();
+            let new_value = self.tcx.fold_regions(
+                value.clone().skip_binder(),
+                &mut false,
+                |r, current_index| {
+                    match *r {
+                        ty::ReLateBound(db, br) if db == current_index => {
+                            let region = region_map.entry(br.kind).or_insert_with(|| {
+                                start_or_continue(&mut self, &mut binders_string, "for<", ", ");
+                                let kind = match br.kind {
+                                    ty::BrNamed(_, name) => {
+                                        do_continue(&mut self, &mut binders_string, name);
+                                        br.kind
+                                    }
+                                    ty::BrAnon(_) | ty::BrEnv => {
+                                        let name = loop {
+                                            let name = name_by_region_index(region_index);
+                                            region_index += 1;
+                                            if !self.used_region_names.contains(&name) {
+                                                break name;
+                                            }
+                                        };
+                                        do_continue(&mut self, &mut binders_string, name);
+                                        ty::BrNamed(DefId::local(CRATE_DEF_INDEX), name)
+                                    }
+                                };
+                                self.tcx.mk_region(ty::ReLateBound(
+                                    ty::INNERMOST,
+                                    ty::BoundRegion { var: br.var, kind },
+                                ))
+                            });
+                            ty::fold::shift_vars(self.tcx, region, current_index.as_u32())
+                        }
+                        ty::RePlaceholder(ty::PlaceholderRegion { name: kind, .. }) => {
+                            let region = region_map.entry(kind).or_insert_with(|| {
+                                let kind = match kind {
+                                    ty::BrNamed(_, name) => {
+                                        start_or_continue(
+                                            &mut self,
+                                            &mut binders_string,
+                                            "for<",
+                                            ", ",
+                                        );
+                                        do_continue(&mut self, &mut binders_string, name);
+                                        kind
+                                    }
+                                    ty::BrAnon(_) | ty::BrEnv => {
+                                        return r;
+                                    }
```

Comment:
So, when I did have this, I was getting some new `for<'r> Send` is not implemented for `X` in some async tests. This might be something to look into, but I figured for now we don't need to worry about it.